docs/src/rules/no-useless-assignment.md

```diff
@@ -0,0 +1,116 @@
+---
+title: no-useless-assignment
+rule_type: suggestion
+related_rules:
+- no-unused-vars
+---
+
+
+
+If the variable is not used after assignment is needless and most likely indicates that there is some behavior that the author wanted but didn't correctly do.
```

Comment:
We definitely need more of a description here. We need to explain the concept of a "dead store" and why it's a problem. We also need to show an example and explain it.